### ISSUE TITLE
Can’t switch audio from TV app

### ISSUE THREAD

**Issue by ai** [jellyfin/jellyfin-tizen#172]
I installed app by [Docker image](https://github.com/Georift/install-jellyfin-tizen).

I tried a few different `.mp4` and `.mkv` files and found a bug with audio switching.

TV app plays only audio matched to app language. I can’t switch audio in both video player and video page. I can change the audio in the selector, but TV will play the default audio.

I tried the same in both web app and Android app and it works there. The problem only in TV app.

There is no problem with switching between subtitles, I found a problem only with audio.

**Comment by dmitrylyzo** [comment]
This is known issue. It should be fixed in the next release (10.8.9).
If your TV is Tizen < 5.5, it will remux - `audioTracks` API doesn't work as expected.

**Comment by ai** [comment]
Nice!

Am I right that I need to re-install app by Docker on release?

How I can subscribe for new releases?

**Comment by dmitrylyzo** [comment]
> Am I right that I need to re-install app by Docker on release?

Yes, if you're using those builds - the app needs to be rebuilt with the new web.

> How I can subscribe for new releases?

[Here](https://github.com/jellyfin/jellyfin/releases) will be a note. Not sure if you can only subscribe to releases.

**Comment by ai** [comment]
@dmitrylyzo I installed the [new version](https://github.com/jeppevinkel/jellyfin-tizen-builds/releases/tag/2023-01-22) but we still have an issue with audio switching.

What debug info I can get you to help?

**Comment by dmitrylyzo** [comment]
What version of Tizen?

Is `Allow video playback that requires conversion without re-encoding` enabled for the user?
_Dashboard / Users / Profile_

**Comment by ai** [comment]
> What version of Tizen?

`Settings` → `Support` → `System update` shows: `Version 1330`

TV Model UE49NU7105

> Is Allow video playback that requires conversion without re-encoding enabled for the

Yes

**Comment by dmitrylyzo** [comment]
> TV Model UE49NU7105

I guess this is Tizen 4 (https://developer.samsung.com/smarttv/develop/specifications/tv-model-groups.html).
Selecting a non-first audio track should result in remuxing (if both video and audio codecs are supported).

How it plays (DirectPlay/Remuxing/DirectStreaming/Transcoding)?
_`Playback Info` in the player or `i` button on the Dashboard_

If it is not `DirectPlay`, please post the ffmpeg log.

**Comment by Vigno04** [comment]
Problem still persist on jellyfin version 10.10 with a q60d 2024 Samsung tv, I can't select language in any way not even changing it in the preferred language work

**Comment by Seltyk** [comment]
Issue continues on 10.10.6, both server and app. My DU7200 can only play anything if I leave the audio language selection as the first item on the list.

- setting preferred language does not affect this
- the audio language played will always be the *second* item in the list
- if there is only one audio language, there will be no audio

If a file has English, French, and Spanish audio listed in that order, English must be selected (or playback will fail) and French audio will be loaded. If a file only has English, there will be no audio. If a file has 7.2 English, then 5.1 English, then whatever else, the 7.2 must be selected and the 5.1 will play (presumably; I don't actually have enough physical audio channels to check that).

**Comment by dmitrylyzo** [comment]
@Seltyk 
1. Language doesn't matter.
2. Make sure `Allow video playback that requires conversion without re-encoding` is enabled for the user (in Dashboard). Tizen 8 requires remuxing for secondary audio.
3. Please post media info of the problematic files. So far I can only guess that the "English" audio requires transcoding, but it's disabled.

**Comment by Seltyk** [comment]
That option does not appear to affect behavior. For what it's worth, I'd like to avoid transcoding if possible; that'd save system resources while guaranteeing maximum output quality. I have all my content as either H.264 or HEVC which *should* be natively supported by my system. Although; transcoding is preferable to non-functionality.

As far as I can tell, all my files are problematic. Here are the `ffmpeg -i` outputs for the [first](https://github.com/user-attachments/files/19041870/en-fr-sp.txt), [second](https://github.com/user-attachments/files/19041871/en-only.txt), and [third](https://github.com/user-attachments/files/19041872/en7-en5.txt) examples respectively.

**Comment by dmitrylyzo** [comment]
> For what it's worth, I'd like to avoid transcoding if possible; that'd save system resources while guaranteeing maximum output quality. I have all my content as either H.264 or HEVC which _should_ be natively supported by my system.

Audio transcoding is much easier than video transcoding and does not affect video quality.
If you want to avoid transcoding completely, convert media files to a fully compatible format (both video and audio).

> I have TrueHD disabled but DTS enabled, as I have a DTS soundbar (Samsung HW-B650).

There's a possibility that the TV doesn't pass-through  a format that isn't natively supported (from the app). So the checkbox may not work.

FYI, when DirectPlay, if the TV doesn't support the first audio track, the second audio track is played. That's why French AC3 is played when English DTS is selected.

> If I disable DTS in the app, no files play regardless of language selection.

For languages in DTS, you probably have `Allow audio playback that requires transcoding` disabled.

> If I disable both DTS in the app and `Allow video playback that requires conversion without re-encoding` in the Dashboard, I get a proper "not supported" playback rejection screen.

In theory, it is supposed to use DirectPlay, and play the first supported audio track. For `en-fr-sp`, French should be played. When there is no supported audio, I don't know, maybe it should refuse to play. 🤷‍♂️

To summarize, you need to enable `Allow video playback that requires conversion without re-encoding` (otherwise the TV will not be able to change the audio track) and `Allow audio playback that requires transcoding` (to re-encode unsupported audio codecs such as DTS).

**Comment by Seltyk** [comment]
All "Allow X playback that requires Y" are enabled on my dashboard. Switching from the TrueHD build to the regular build just magically makes the language problem go away??? Weird. Other issues I experience are out of the scope of this ticket. Thanks for the help regardless.
